packages/jazz-react/src/index.tsx

```diff
@@ -126,12 +126,12 @@ export function createJazzReactApp<Acc extends Account>({
   }
 
   function useAccount(): { me: Acc; logOut: () => void };
-  function useAccount<D extends DepthsIn<Acc>>(
-    depth: D,
-  ): { me: DeeplyLoaded<Acc, D> | undefined; logOut: () => void };
-  function useAccount<D extends DepthsIn<Acc>>(
-    depth?: D,
-  ): { me: Acc | DeeplyLoaded<Acc, D> | undefined; logOut: () => void } {
+  function useAccount<const O extends { resolve?: RefsToResolve<Acc> }>(
+    options: O,
+  ): { me: Resolved<Acc, O> | undefined; logOut: () => void };
```

Comment:
This kind of typing makes ok to add new options like:

<img width="522" alt="Screenshot 2024-12-17 at 16 17 56" src="https://github.com/user-attachments/assets/842f2eaa-4954-4ef5-b36e-422ae476bf71" />

Do you think it would be ok to have just:

```ts
useAccount<const R extends RefsToResolve<Acc>>(
    options: { resolve?: R },
  ): { me: Resolved<Acc, R> | undefined; logOut: () => void };
```

Reply:
Also noticed that with this change it would be possible to add docs on the resolve option:
<img width="855" alt="Screenshot 2024-12-17 at 16 23 34" src="https://github.com/user-attachments/assets/05f821c6-0fd9-45c2-9e91-1c700e04d114" />

Reply:
Noticed two other issues:
1. Additional properties added inside to resolve aren't marked as invalid
1. Errors on the value side are quite cryptic

We probably can't do much to make errors easier to read, but we should definitely mark additional properties as errors.

See:
<img width="1075" alt="Screenshot 2024-12-17 at 16 38 31" src="https://github.com/user-attachments/assets/a2e1143b-32dc-42eb-955e-847d65995e7f" />